### PR TITLE
Swift3 conversion

### PR DESCRIPTION
**Pull request by greencoder**
We are converting our reference apps to Swift 3 and iOS 10.

### CODE REVIEW

File: CampaignKitReference/AppDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  content.categoryIdentifier = "com.radiusnetworks.localNotification"
+  
+  let trigger = UNTimeIntervalNotificationTrigger.init(timeInterval: 0.25, repeats: false)
+  let request = UNNotificationRequest.init(identifier: "CKNotification", content: content, trigger: trigger)
+
+  let center = UNUserNotificationCenter.current()
+  center.add(request)
+  
+  print("Added request")
+}
```
Comment (mharper) [inline CampaignKitReference/AppDelegate.swift]: Is there a reason this function is in global scope?

File: CampaignKitReference/ViewController.swift [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  override func viewWillDisappear(_ animated: Bool) {
+    super.viewWillDisappear(animated)
+    NotificationCenter.default.removeObserver(self)
+  }
+
+  // MARK: - Notification Handler
+  
+  func campaignFound(notification: NSNotification) {
+    if let userInfo = notification.userInfo {
+      if let campaign = userInfo["campaign"] as? CKCampaign {
```
Comment (mharper) [inline CampaignKitReference/ViewController.swift]: You can avoid the "pyramid of death" by replacing this construct with:
```
if let userInfo = notification.userInfo, let campaign = userInfo["campaign"] as? CKCampaign { ...
```

File: CampaignKitReference/ViewController.swift [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      }
+    }
+  }
+
+  // MARK: - User Interface Handling
+  
+  func showDeal(_ content: String?) {
+    let storyboard = UIStoryboard(name: "Main", bundle: nil)
+    let navVC = storyboard.instantiateViewController(withIdentifier: "DealNavController") as! UINavigationController
+    let dealVC = navVC.childViewControllers.first as! DealViewController
```
Comment (mharper) [inline CampaignKitReference/ViewController.swift]: _Probably_ should be `if-let` instead of `!` just to be a good role model.

**Review by mharper — APPROVED** [review]
LGTM!

File: CampaignKitReference/AppDelegate.swift (hunk 1 above)
Comment (greencoder) [inline CampaignKitReference/AppDelegate.swift]: Not really, only that I was trying to keep the delegate method a little cleaner. Do you have a suggestion?

File: CampaignKitReference/ViewController.swift (hunk 2 above)
Comment (greencoder) [inline CampaignKitReference/ViewController.swift]: The compiler would not let me cast it as a `CKCampaign` in the `if` statement.

Comment (greencoder) [inline CampaignKitReference/ViewController.swift]: Somehow I had the syntax wrong when I tried this. Thanks - yours works.

File: CampaignKitReference/ViewController.swift (hunk 3 above)
Comment (greencoder) [inline CampaignKitReference/ViewController.swift]: screen shot 2016-09-30 at 12 27 28 pm

Comment (greencoder) [inline CampaignKitReference/ViewController.swift]: It doesn't like it.

**Comment by greencoder** [comment]
@csexton Are you okay with me merging this into master?

**Review by csexton — APPROVED** [review]
So swifty! 🚴 LGTM

**Pull request merged**